src/lib/utils/profile.ts

```diff
@@ -1,83 +1,37 @@
 import { subgraphRequest } from '@/lib/utils/subgraph';
+import getProvider from '@/lib/utils/provider';
 
-function get3BoxProfiles(addresses) {
-  return new Promise((resolove, reject) => {
-    subgraphRequest('https://api.3box.io/graph', {
+async function get3BoxProfile(address: string): Promise<Record<any, any>> {
+  try {
+    const { profiles: [profile] } = await subgraphRequest('https://api.3box.io/graph', {
       profiles: {
         __args: {
-          ids: addresses
+          ids: [address]
         },
         name: true,
         eth_address: true,
         image: true
       }
-    })
-      .then(({ profiles }) => {
-        const _3BoxProfiles = {};
-        profiles.forEach(profile => {
-          _3BoxProfiles[profile.eth_address.toLowerCase()] = profile;
-        });
-        resolove(_3BoxProfiles);
-      })
-      .catch(error => {
-        reject(error);
-      });
-  });
-}
-
-function lookupAddresses(addresses) {
-  return new Promise((resolove, reject) => {
-    subgraphRequest('https://api.thegraph.com/subgraphs/name/ensdomains/ens', {
-      accounts: {
-        __args: {
-          first: 1000,
-          where: {
-            id_in: addresses.map(addresses => addresses.toLowerCase())
-          }
-        },
-        id: true,
-        domains: {
-          __args: {
-            first: 1
-          },
-          name: true,
-          labelName: true
-        }
-      }
-    })
-      .then(({ accounts }) => {
-        const ensNames = {};
-        accounts.forEach(profile => {
-          ensNames[profile.id.toLowerCase()] = profile?.domains?.[0]?.labelName
-            ? profile.domains[0].name
-            : '';
-        });
-        resolove(ensNames);
-      })
-      .catch(error => {
-        reject(error);
-      });
-  });
+    });
+    return profile;
+  } catch (error) {
+    console.error(error);
+    return {}
+  }
 }
 
-export async function getProfiles(addresses) {
-  let ensNames: any = {};
-  let _3BoxProfiles: any = {};
+export async function getProfile(address: string, network: string) {
   try {
-    [ensNames, _3BoxProfiles] = await Promise.all([
-      lookupAddresses(addresses),
-      get3BoxProfiles(addresses)
+    const provider = getProvider(network);
+    const [ensName, _3BoxProfile] = await Promise.all([
+      provider.lookupAddress(address),
```

Comment:
Based on the [ENS docs](https://docs.ens.domains/dapp-developer-guide/resolving-names#reverse-resolution), it looks like this is all that's needed for a reverse resolution.